Ginger/GingerCoreCommon/Repository/SolutionRepositoryItemInfo.cs

```diff
@@ -164,5 +195,27 @@ RepositoryFolder<T> GetRepositoryFolderParent(RepositoryFolderBase repositoryFol
 
             return null;
         }
+
+        internal void AllItemsCacheRemove(T item)
+        {
+            mAllItemsCache.Remove(item);
```

Comment:
don't we need to Lock also the Remove operation so 2 Threads won't try to delete the same item and one of them will fail? 
i think we need to add Lock and before Remove check if item still exist on List